internal/prometheusbpint/modules.go

```diff
@@ -0,0 +1,31 @@
+package prometheusbpint
+
+import (
+	"runtime/debug"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var goModules = promauto.With(GlobalRegistry).NewGaugeVec(prometheus.GaugeOpts{
+	Name: "baseplate_go_modules",
+	Help: "Export the version information for included Go modules, and whether the module is the 'main' module or a 'dependency'.  Always 1",
+}, []string{"go_module", "module_role", "module_replaced", "module_version"})
+
+func RecordModuleVersions(info *debug.BuildInfo) {
+	record := func(role string, mod *debug.Module) {
```

Comment:
Out of curiosity, what's the reason for defining this inside rather than at the top-level?